### PR TITLE
feat: Added `escape_regex` operation to the `str` namespace and as a global function

### PR DESCRIPTION
**Pull request by barak1412**
Fixes https://github.com/pola-rs/polars/issues/19207.

### CODE REVIEW

**Comment by barak1412** [comment]
@orlp Just to be sure -

The code:

```python
import polars as pl

df = pl.DataFrame({"text": ["abc", "def", None, "abc(\\w+)"]})

df.with_columns(escaped=pl.escape_regex('text'))
```

Should escape the `'text'` literal, not the column, right?

Besides, I need to:
1. Add tests.
2. Add `pl.escape_regex` as function.
3. Refactor the code such that the namespace `Expr` will be translated to the function `Expr`.

**Comment by orlp** [comment]
@barak1412 `pl.escape_regex` should *only* work on Python strings, and not interact with the expression API at all. I would suggest adding a warning/error if you try to pass in an expression into it, suggesting you to use `Expr.str.escape_regex` instead.

File: crates/polars-ops/src/chunked_array/strings/escape_regex.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+use polars_core::prelude::arity::unary_elementwise;
+use polars_core::prelude::StringChunked;
+use regex::escape;
+
+#[inline]
+pub fn escape_regex_str(s: &str) -> String {
```
Comment (barak1412) [inline crates/polars-ops/src/chunked_array/strings/escape_regex.rs:6]: Added this function so `pl.escape_regex` and `str.escape_regex` will be coupled by same implementation.

**Comment by barak1412** [comment]
@orlp I will be glad if you can look, thanks.

File: crates/polars-ops/src/chunked_array/strings/escape_regex.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    escape(s)
+}
+
+fn escape_regex_helper(s: Option<&str>) -> Option<String> {
+    s.map(escape_regex_str)
+}
+
+pub fn escape_regex(ca: &StringChunked) -> StringChunked {
+    unary_elementwise(ca, escape_regex_helper)
```
Comment (orlp) [inline crates/polars-ops/src/chunked_array/strings/escape_regex.rs:15]: Can you add a dependency on `regex_syntax` (we already have this indirectly), and use a loop similar to this?

```rust
pub fn escape_regex(ca: &StringChunked) -> StringChunked {
    let mut buffer = String::new();
    let mut builder = StringChunkedBuilder::new(ca.name().clone(), ca.len());
    for opt_s in ca.iter() {
        if let Some(s) = opt_s {
            buffer.clear();
            regex_syntax::escape_into(s, &mut buffer);
            builder.append_value(&buffer);
        } else {
            builder.append_null();
        }
    }
}
```

This prevents us from having to allocate for each string.

File: py-polars/polars/functions/escape_regex.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    Escapes string regex meta characters.
+
+    Parameters
+    ----------
+    s
+        The string that all of its meta characters will be escaped.
+
+    """
+    if isinstance(s, pl.Expr):
+        msg = "escape_regex function is unsupported for `Exp`, you may want use `Expr.str.escape_regex` instead"
```
Comment (orlp) [inline py-polars/polars/functions/escape_regex.py:21]: Typo: `Exp` -> `Expr`.

**Review by orlp — CHANGES_REQUESTED** [review]

**Comment by barak1412** [comment]
@orlp Should be the right fix now, thanks.

**Review by orlp — APPROVED** [review]

**Comment by ritchie46** [comment]
Looks great @barak1412, we only miss docs entries in the python reference guide. Then it is good to go.

**Comment by barak1412** [comment]
@ritchie46 Sure, added to the docs.

I hope I changed the right places, first time I am touching the docs.

**Comment by ritchie46** [comment]
Thanks @barak1412

**Pull request merged**